### PR TITLE
Improve coverage reporting

### PR DESCRIPTION
**Pull request by bdice**
## Description
This PR improves coverage reporting configurations to ignore external modules (e.g. tqdm, six).

## Types of Changes

- [x] Miscellaneous

This pull request is based on
- [x] *develop*, because it introduces a new feature.

## Checklist:
- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac/blob/master/contributors.txt).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [x] I have updated the [changelog](https://github.com/glotzerlab/signac/blob/master/changelog.txt).

### CODE REVIEW

**Comment by bdice** [comment]
@csadorf Here is our current coverage report on codecov: https://codecov.io/gh/glotzerlab/signac/branch/misc%2Fimprove-coverage-reporting

File: changelog.txt [hunk 1]
```diff
@@ -31,6 +31,7 @@ Added
  - Enable deleting of `JSONDict` keys through the attribute interface, e.g., `del job.doc.foo`.
  - The `Job.init()` method returns the job to allow one-line job creation and initialization.
  - Add HTML representation of instances of `Project` and `JobsCursor` targeted at Jupyter Notebooks (requires pandas).
+ - Code coverage is now uploaded to https://codecov.io/gh/glotzerlab/signac/
```
Comment (csadorf) [inline changelog.txt]: I don't think that's something we should announce in the changelog.

Comment (bdice) [reply changelog.txt]: Sorry, I'll retract this commit. Just checking the boxes on the PR form. ;)

Comment (bdice) [reply changelog.txt]: Sorry, I'll retract this commit. Just checking the boxes on the PR form. ;)

**Pull request merged**